File: src/subtractProductAndSum.js
// LeetCode # 1281
// runtime: 80ms top 52.85%
// memory usage: 38.8mb top 44.95%
// https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/
export default function subtractProductAndSum(num) {
  let array = [...num.toString()];
  return (
    array.reduce((a, b) => parseInt(a) * parseInt(b)) -
    array.reduce((a, b) => parseInt(a) + parseInt(b))
  );
}
console.log(subtractProductAndSum(234));

File: src/subtract_product_and_sum.rs
/// LeetCode # 1281
/// https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/

pub fn subtract_product_and_sum(num: i32) -> i32 {
    let integer = 32i32;
    let string = num.to_string();
    // let array:Vec<i32> = string.parse.collect();
    println!("{:?}", string);
    println!("{:?}", string.len());
    integer
}

#[cfg(test)]
mod tests {
    use super::subtract_product_and_sum;
    #[test]
    fn test_subtract_product_and_sum() {
        // assert_eq!(subtract_product_and_sum(234), 15);
        // assert_eq!(subtract_product_and_sum(4421), 1);
        assert_eq!(subtract_product_and_sum(4421), 32);
    }
}
